generator/format_utils.cpp

```diff
@@ -0,0 +1,62 @@
+#include "format_utils.hpp"
+
+#include <algorithm>  // min
+#include <limits>  // size_t::max
+#include <numeric>  // accumulate
+#include <range/v3/all.hpp>
+
+
+namespace tree_gen::cpp {
+
+/**
+ * Formats string literal containing C++ code.
+ *
+ * string_literal is the input string.
+ * indent is the indentation, in number of spaces, of the output string.
+ * remove_first_new_line will be true if we want the first new line of the input string to be removed, false otherwise.
+ */
+std::string format_string_literal(std::string_view string_literal, size_t indent, bool remove_first_line) {
```

Comment:
As dicussed would be nice to do some preprocessing here to avoid having to escpae { as {{ all the time :)

Reply:
Done. I can't believe this took me all day haha.